second-analysis-steps/analysis-automation-snakemake.md

```diff
@@ -0,0 +1,221 @@
+# Analysis automation with snakemake
+
+{% objectives "Learning Objectives" %}
+* Objective one
+* Objective two
+{% endobjectives %}
+
+> “The Snakemake workflow management system is a tool to create reproducible and scalable data analyses”
+
+## Motivation
+
+* A workflow management system allows you to:
+  * Keep a memory of how your scripts are used
+  * Keep a memory of where your inputs are stored
+  * Run multiple steps in sequence
+  * Automatically detect if something changed
+* A workflow management forces you to:
+  * Keep your code and your locations in order
+  * Structure your code so that it is user-independent
+  * Standardise your scripts
+  * Bonus: Standardised scripts can sometimes be used across analyses
+
+## Snakemake: Documentation and installation
+
+You can find full documentation on Snakemake [at this link](http://snakemake.readthedocs.io/en/stable/index.html).
+
+You can find lines to install snakemake on linux in the file [at this link](https://github.com/lhcb/starterkit-lessons/blob/snakemake/second-analysis-steps/code/snakemake/install_snake.sh).
```

Comment:
If we suggest using conda we should use Python 3.6 (bioconda doesn't support 3.4 anymore).

That said I think it would be better to use a LCG Python release instead (like in [analysis-essentials](https://lhcb.github.io/analysis-essentials/python/running.html)) that uses `pip install --user` and adds a `snakemake` function to their `.*rc` file. My suggestion would be:

```bash
curl -L http://cern.ch/go/l8fT | bash
```

Reply:
I won't have time to test this before the 10th. But I'll keep it in mind for the future.

Reply:
The Impactkit/Starterkit lessons generally try to have the students follow along and do mini exercises where possible. Do you have any plans to do this?

If time to prepare is a problem, I can try to edit this PR to use LCG and have some specific exercises if you want?

Reply:
The exercises are already there. You already put "{% challenge" tags around them. What do you mean with specific? For LCG, yes, if you want feel free to modify so that it will use it. But please leave the old installation file there at least until we are sure everything works.

Reply:
Well this was intended as a live tutorial. So I would show what I mean with "grep". But ok you can change it the way you think it's more useful.

Reply:
It's useful for people that use the tutorials without attending the workshop for whatever reason, I'll see what I can do over the next few of days.